app/forms/forms/type_of_answer_form.rb

```diff
@@ -0,0 +1,21 @@
+class Forms::TypeOfAnswerForm
+  include ActiveModel::Model
+  include ActiveModel::Validations
+
+  attr_accessor :answer_type, :form
+
+  validates :answer_type, presence: true, inclusion: { in: Page::ANSWER_TYPES }
+
+  def submit(session)
+    return false if invalid?
+
+    session[:page] = { answer_type: }
+  end
+
+  def number(form)
+    # If this page is in form, return the position, else it must be new so
+    # return the number if it was inserted at the end
+    index = form.pages.index(self)
+    (index.nil? ? form.pages.length : index) + 1
+  end
```

Comment:
`number(form)` is now in two places. - /models/Page.rb and here. Could we put it somewhere better? would it make more sense in /models/Form.rb and taking page as an argument?

Reply:
This has now been refactored into `/models/Form.rb`, as well as flushing out a bug with the page number caption for editing the answer type!